language-support/hs/bindings/src/DA/Ledger/Types.hs

```diff
@@ -187,17 +188,23 @@ data Value
     | VContract ContractId
     | VList [Value]
     | VInt Int
-    | VDecimal Text -- TODO: Maybe use Haskell Decimal type
-    | VString Text
-    | VTimestamp MicroSecondsSinceEpoch
+    | VDecimal Text
+    | VText Text
+    | VTime MicroSecondsSinceEpoch
     | VParty Party
     | VBool Bool
     | VUnit
     | VDate DaysSinceEpoch
     | VOpt (Maybe Value)
     | VMap (Map Text Value)
+    | VEnum Enum
     deriving (Eq,Ord,Show)
 
+data Enum = Enum
+    { eid   :: Maybe Identifier
```

Comment:
I assume this field contains the name of the enum type. If so, we should use a better name, like `enumType`.

Reply:
I was wondering if we could get rid of these fields entirely.
They are optional in the API.
When sending they can always be omitted
When receiving, they only get filled in in `verbose` mode.
I don't really see what purpose they serve.

Reply:
They make the Ledger API nominally typed instead of "just" structurally typed. For example, they are good for sanity checking.